core/templates/cowdata.h

```diff
@@ -64,45 +64,74 @@ class CowData {
 	template <class TV, class VV>
 	friend class VMap;
 
+public:
+#ifdef IS_32_BIT
+	typedef int32_t Size;
+	typedef uint32_t USize;
+#else
+	typedef int64_t Size;
+	typedef uint64_t USize;
```

Comment:
Might need more specific names, I'm worried about conflicts with potential Windows or macOS global defines.

Reply:
It's in the local class context so it should be fine as long as it passes CI.

Reply:
Is this extra complexity really necessary. 32-bit systems won't benefit for it, but using 64-bit ints should work fine on 32-bit systems (as long as it's not cast to `void*` directly).

Reply:
And if we want to do it this way for some reason, why not use `size_t` and `ptrdiff_t` directly?

Reply:
I don't mind leaving it 64 bits even for 32 bits systems.